base/rational.jl

```diff
@@ -156,6 +156,9 @@ widen(::Type{Rational{T}}) where {T} = Rational{widen(T)}
 Approximate floating point number `x` as a [`Rational`](@ref) number with components
 of the given integer type. The result will differ from `x` by no more than `tol`.
 
+!!! note
+    If `x` is an `Irrational` number, then `T` is a required argument.
```

Comment:
Shouldn't we just define the method to be consistent rather than document the exception?

Reply:
I believe we shouldn't define it to be consistent since `Irrational` types has their own story to tell. `Irrational` values as know take up their types from explicit conversion or mathematical operations, so what type should `rationalize` convert it to?

So you see, I think the core devs thought this out and just felt its best if the user explicitly stated the type they want the result to be translated to, since `Irrational` types have no defined types of their own.

Reply:
> `Irrational` values as know take up their types from explicit conversion or mathematical operations, so what type should `rationalize` convert it to?

`Int`, like for all AbstractFloats. I think there would be an argument here not to define it if the default conversion type as precision dependent, but since it's not, I don't see an argument not to just define it.

Reply:
Well whichever we choose to go, its okay. But for now, I think its best to add the "!!! note""" and then decide later if the `rationalize` would implicitly convert to `Int`, something I don't think is good though, since `Irrational` values are floating-points in most cases.

Reply:
`Irrational` values are placeholders meant to be converted to a desired type. I don't see anything inconsistent or unexpected about defaulting to an `Int` representation.